src/components/Menu/Menu.tsx

```diff
@@ -49,33 +55,60 @@ export type MenuItemsProps = ExtractProps<typeof HeadlessMenu.Items>;
  *
  * A dropdown that reveals or hides a list of actions
  */
-export const Menu = ({ className, ...other }: MenuProps) => {
+export const Menu = ({
+  buttonText,
+  children,
+  className,
+  ...other
+}: MenuProps) => {
   const menuClassNames = clsx(className, styles['menu']);
-  return <HeadlessMenu as="div" className={menuClassNames} {...other} />;
+  if (typeof children === 'function') {
+    return (
+      <HeadlessMenu as="div" className={menuClassNames} {...other}>
+        {({ open, close }) => (
+          <>
+            {buttonText && (
```

Comment:
Is `buttonText` mostly a more convenient way to provide a custom button?

Reply:
I agree, and I think I over complicated this lol.
How about: keeping the changes from ClickableStyle to Button, but also keeping the changes to accept Button props. I can add a conditional prop for the chevron icon. This way, the changes won't be breaking, but the Menu.Button will still have more customizability

Reply:
I dig it